File: src/main/java/com/huythanh0x/training_thanhvh_java_spring_jwt_jpa/repository/UserRepository.java
package com.huythanh0x.training_thanhvh_java_spring_jwt_jpa.repository;


import com.huythanh0x.training_thanhvh_java_spring_jwt_jpa.model.user.UserEntity;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserRepository extends JpaRepository<UserEntity, Integer> {
    Optional<UserEntity> findByUsername(String username);

    Boolean existsByUsername(String username);
}

File: modules/coupon-domain/src/main/kotlin/com/thanh0x/coursedeal/repository/UserRepository.kt
package com.thanh0x.coursedeal.repository

import com.thanh0x.coursedeal.model.user.UserEntity
import org.springframework.data.jpa.repository.JpaRepository

interface UserRepository : JpaRepository<UserEntity, Int> {
    fun findByUsername(username: String): UserEntity?

    fun existsByUsername(username: String): Boolean

    fun findByEmail(email: String): UserEntity?

    fun existsByEmail(email: String): Boolean
}
